src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -476,45 +476,101 @@ private NewCheckRun CheckRunForAdd(MergePolicyEvaluationResult.SingleResult resu
                 throw new ArgumentNullException(nameof(result.MergePolicyName));
             }
 
-            var newCheck = new NewCheckRun(result.MergePolicyDisplayName, sha);
+            var newCheckRun = new NewCheckRun(result.MergePolicyDisplayName, sha);
+            newCheckRun.ExternalId = CheckRunId(result.MergePolicyName, sha);
+            UpdateCheckRun(newCheckRun, result);
+            return newCheckRun;
+        }
+
+        /// <summary>
+        ///     Update a check run based on a NewCheckRun and evaluation
+        /// </summary>
+        /// <param name="newCheckRun">The NewCheckRun that needs to be updated</param>
+        /// <param name="eval">The result of that updated check run</param>
+        /// <returns>The updated CheckRun</returns>
+        private CheckRunUpdate CheckRunForUpdate(CheckRun checkRun, MergePolicyEvaluationResult.SingleResult eval)
+        {
+            if (eval.MergePolicyName == null)
+            {
+                throw new ArgumentNullException(nameof(eval.MergePolicyName));
+            }
+
+            CheckRunUpdate updatedCheckRun = new CheckRunUpdate();
+            updatedCheckRun.Name = checkRun.Name;
+            updatedCheckRun.ExternalId = checkRun.ExternalId;
+            UpdateCheckRun(updatedCheckRun, eval);
+            return updatedCheckRun;
+        }
+
+        /// <summary>
+        ///     Create a CheckRunUpdate based on a check run that needs to be deleted
+        /// </summary>
+        /// <param name="checkRun">The check run that needs to be deleted</param>
+        /// <returns>The deleted check run</returns>
+        private CheckRunUpdate CheckRunForDelete(CheckRun checkRun)
+        {
+            CheckRunUpdate updatedCheckRun = new CheckRunUpdate();
+            updatedCheckRun.Name = checkRun.Name;
+            updatedCheckRun.ExternalId = checkRun.ExternalId;
```

Comment:
In this case you don't need to set properties which are not gonna change. I believe that null values from CheckRunUpdate are not changed. Can you please validate it on your prove of concept?